### PR TITLE
[RecentMacros] Check and warn for shortcut conflicts + small fixes/improvements

### PR DESCRIPTION
**Pull request by 0penBrain**
https://forum.freecadweb.org/viewtopic.php?f=3&t=65081#p564743

Set as Draft time it's discussed and tested.
@mwganson may you have a look ?

### CODE REVIEW

File: src/Gui/Action.cpp [hunk 1]
```diff
@@ -921,15 +921,26 @@ void RecentMacrosAction::setFiles(const QStringList& files)
...
         if (index < shortcut_count){
-            recentFiles[index]->setShortcut(accel);
+            auto accel_tmp = QString::fromStdString(shortcut_modifiers);
+            accel_tmp.append(QString::number(index+1,10)).toStdString();
+            auto check = Application::Instance->commandManager().checkAcceleratorForConflicts(qPrintable(accel_tmp));
+            if (check) {
+                recentFiles[index]->setShortcut(QKeySequence());
+                auto msg = QLatin1String("Recent macros : shortcut %1 disabled because conflicting with %2\n")
+                                                            .arg(accel_tmp).arg(QLatin1String(check->getName()));
+                Base::Console().Warning(qPrintable(msg));
```
Comment (wwmayer) [inline src/Gui/Action.cpp:936]: DON'T do this! For more details see: https://owasp.org/www-community/attacks/Format_string_attack
Instead write:
`Base::Console().Warning("%s", qPrintable(msg));`

Comment (0penBrain) [reply src/Gui/Action.cpp]: Fixed. Thanks.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5437">pipeline status</a> for feature branch [PR_5437](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5437). Pipeline [456257545](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/456257545) was triggered at [6baa431](https://github.com/FreeCAD/FreeCAD/pull/5437/commits/6baa43146f007906df81bf6f867bb39fbf0d346b). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by 0penBrain** [comment]
Just pushed a new version because CI caught that QLatin1String::arg() was only introduced in Qt 5.14.
Replaced with QStringLiteral::arg().

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5437">pipeline status</a> for feature branch [PR_5437](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5437). Pipeline [457077919](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/457077919) was triggered at [ce1f617](https://github.com/FreeCAD/FreeCAD/pull/5437/commits/ce1f6172fbd8b6551ea26af26fba95960aff294b). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5437">pipeline status</a> for feature branch [PR_5437](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5437). Pipeline [462996933](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/462996933) was triggered at [4adeca5](https://github.com/FreeCAD/FreeCAD/pull/5437/commits/4adeca5e1af1f6294866966d79722b40c85de0ec). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Pull request merged**